src/components/router/routerHelpers.js

```diff
@@ -29,6 +51,100 @@ const baseName = (helpers.TEST_MODE && '/') || (helpers.DEV_MODE && '/') || dyna
 const basePath =
   (helpers.TEST_MODE && '/') || (helpers.DEV_MODE && '/') || window.location.pathname.split(helpers.UI_NAME)[0];
 
+/**
+ * ToDo: now that navigation is generated we can bypass and use route configuration directly
+ * Review removing "navigation" and using the route config directly.
+ */
+/**
+ * Generate navigation list, used to help in look ups for "missing products" and provide additional route details.
+ *
+ * @param {Array} config
+ * @returns {{path: string, default: boolean, component: Node, productParameter: string,
+ *     viewParameter: string, activateOnError: boolean, id: string, isSearchable: boolean, pathParameter: string}[]}
+ */
+const generateNavigation = (config = routesConfig) =>
+  config.map(({ pathParameter, productParameter, ...params }) => ({
+    pathParameter: (Array.isArray(pathParameter) && pathParameter[0]) || pathParameter,
+    productParameter: (Array.isArray(productParameter) && productParameter[0]) || productParameter,
+    viewParameter:
+      (productParameter && `view${(Array.isArray(productParameter) && productParameter[0]) || productParameter}`) ||
+      productParameter,
+    ...params
+  }));
+
+/**
+ * Return an array of objects that describes navigation against API product IDs.
+ *
+ * @returns {Array}
+ */
+const navigation = generateNavigation();
+
+/**
+ * Generate product groups for applying query filter resets.
+ *
+ * @param {Array} config
+ * @returns {Array}
+ */
+const generateProductGroups = (config = routesConfig) => {
+  const productGroups = {};
+
+  config.forEach(({ pathParameter, productParameter }) => {
+    const viewIds = ((Array.isArray(productParameter) && productParameter) || [productParameter]).map(
+      id => (id && `view${id}`) || id
+    );
+
+    viewIds.forEach((id, index) => {
+      if (id) {
+        if (!productGroups[id]) {
+          productGroups[id] = [];
+        }
+
+        if (pathParameter) {
+          productGroups[id].push((Array.isArray(pathParameter) && pathParameter?.[index]) || pathParameter);
+        }
+      }
+    });
+  });
+
+  return productGroups;
+};
+
+/**
+ * Reference for products grouped by view.
+ */
+const productGroups = generateProductGroups();
+
+/**
+ * Generate routes to be consumed by router.
+ *
+ * @param {Array} config
+ * @returns {Array}
+ */
+const generateRoutes = (config = routesConfig) => {
+  const updatedRoutes = [];
+
+  config.forEach(({ activateOnError, component, disabled, path: routePath, redirect }) =>
```

Comment:
required: now that the condition is removed we can turn this into a direct `=> config.map`